### ISSUE TITLE
When available, FilterSubmitButton should use the button class name provided by the elements API

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#6962]
Currently, the `FilterSubmitButton` component uses the [`wp-block-button__link` class name](https://github.com/woocommerce/woocommerce-blocks/blob/fbf2a2895efb9d3f7a8ee3264ccb107e72f1ea04/assets/js/base/components/filter-submit-button/index.tsx#L33) to get styling from the theme. However, I don't think that class is intended to be used like that; I think it's a class specific to the button block.

As an alternative approach, Gutenberg is introducing the elements API (see https://github.com/WordPress/gutenberg/pull/41753 and pdnLyh-1kY-p2), which will allow us to get a generic button class name that will apply theme styles.

In other words, I think we should check if `@wordpress/block-editor` includes the `__experimentalGetElementClassName()` function (that happens if the store has Gutenberg installed or is in WP >=6.1). If it does, we should get the class returned by `__experimentalGetElementClassName( 'button' )` and use that one instead of `wp-block-button__link`.

* (Alternatively, we could pass the class from PHP so we don't need to import `block-editor` in the frontend)

What do you think @albarin @dinhtungdu?

**Comment by dinhtungdu** [comment]
> However, I don't think that class is intended to be used like that; I think it's a class specific to the button block.

I agree.

> As an alternative approach, Gutenberg is introducing the elements API (see [WordPress/gutenberg#41753](https://github.com/WordPress/gutenberg/pull/41753) and pdnLyh-1kY-p2), which will allow us to get a generic button class name that will apply theme styles.

That function will return `wp-element-button` for `button`. Here is the result when I swapped the `wp-block-button__link` by `wp-element-button`:

![image](https://user-images.githubusercontent.com/5423135/186066978-4a7a17ab-45c0-44d9-bc2c-b81287bdf50f.png)

At this point, I think we have another question to answer: which button style should we use for filter buttons? Generic style or button link style?

The `__experimentalGetElementClassName( 'button' )` returns only `wp-element-button`, so we still need to pass `wp-block-button__link` to it if we want to use the button link style. See [the comment submit button](https://github.com/WordPress/gutenberg/pull/41753/files#diff-612a231ad91e04fbaa4282dc72d65dc74483ac326c7b84ab7946a70f387d97a6R81).

> * (Alternatively, we could pass the class from PHP so we don't need to import `block-editor` in the frontend)

I prefer this because `block-editor` shouldn't be used on the frontend.

**Comment by Aljullu** [comment]
> The `__experimentalGetElementClassName( 'button' )` returns only `wp-element-button`, so we still need to pass `wp-block-button__link` to it if we want to use the button link style. See [the comment submit button](https://github.com/WordPress/gutenberg/pull/41753/files#diff-612a231ad91e04fbaa4282dc72d65dc74483ac326c7b84ab7946a70f387d97a6R81).

Oh, good point, I didn't notice Gutenberg sets `wp-block-button__link` as well. I'm a bit confused by this, but it makes sense being aligned with them.

Given that the issue is now about making use of the Elements API without doing any other changes, I'm closing in favor of #6567.

Thanks for your feedback, @dinhtungdu!
